core/embed/trezorhal/i2c.c

```diff
@@ -1,39 +1,49 @@
 
 
 #include STM32_HAL_H
+#include TREZOR_BOARD
 #include "i2c.h"
 #include "common.h"
 
 static I2C_HandleTypeDef i2c_handle;
 
 void HAL_I2C_MspInit(I2C_HandleTypeDef *hi2c) {
   // enable I2C clock
-  __HAL_RCC_I2C1_CLK_ENABLE();
-  // GPIO have already been initialised by touch_init
+  I2C_INSTANCE_CLK_EN();
```

Comment:
Disco board uses I2C3, instead of I2C1 as model T. So this difference was outsourced to `trezorhal/boards/stm32f429i-disc1.h `and `trezorhal/boards/trezor_t.h`